include/roaring/roaring64.h

```diff
@@ -541,6 +548,12 @@ size_t roaring64_bitmap_portable_deserialize_size(const char *buf,
  * order. This is is guaranteed to happen when serializing an existing bitmap,
  * but not for random inputs.
  *
+ * You may use roaring64_bitmap_internal_validate to check the validity of the
+ * bitmap prior to using it.
+ *
+ * We recommend that you use checksums to check that serialized data corresponds
+ * to a serialized bitmap.
+ *
```

Comment:
Is this a weakening of `roaring64_bitmap_internal_validate`'s guarantees?

At least for rust bindings, I'd like to be able to guarantee that:

* `roaring{,64}_bitmap_internal_validate` will never lead to Undefined Behavior (segfaults, etc) when called on a bitmap returned from a `deserialize_safe` function
* If `roaring{,64}_bitmap_internal_validate` returns true on a bitmap returned from a `deserialize_safe` function, that bitmap is _valid_ and no valid use of that bitmap can lead to Undefined Behavior.

Reply:
@Dr-Emann We never made these guarantees.

Reply:
@Dr-Emann I invite strengthening, review and additional testing.

Reply:
At least, as of now, there is some testing with random inputs. There wasn't previously. :-)

Reply:
There has been at least a fair bit of [fuzzing](https://github.com/RoaringBitmap/croaring-rs/blob/master/fuzz/fuzz_targets/deserialize.rs) with sanitizers I've done from the rust side on `Bitmap{,64}::try_deserialize`, which internally calls deserialize_safe and internal_validate.

Reply:
Oh, and one other guarantee I'd like to be able to make:

* Calling `roaring{,64}_bitmap_free`/`roaring_bitmap_clear` on a bitmap returned from a `deserialize_safe` function - even if the bitmap would return false from `internal_validate` - will not lead to undefined behavior or memory leaks.

Reply:
> There has been at least a fair bit of [fuzzing](https://github.com/RoaringBitmap/croaring-rs/blob/master/fuzz/fuzz_targets/deserialize.rs) with sanitizers I've done from the rust side on Bitmap{,64}::try_deserialize, which internally calls deserialize_safe and internal_validate.

That's great. I was not aware of this work.

Reply:
Turns out, I'm not sure I ever actually ran it after adding 64 bit bitmaps to it.

Reply:
In any case, before we make any claim in CRoaring, we need strong testing. :-)